File: CoreJava/Shoppingcart-app/src/com/techlabs/shoppingcart/Order.java
package com.techlabs.shoppingcart;

import java.text.*;
import java.util.*;

public class Order {
	private int orderid;
	private Date orderDate;
	private List<LineItem> items = new ArrayList<LineItem>();
	SimpleDateFormat sdf = new SimpleDateFormat("DD/MM/YY");
	
	public Order(int orderid, Date orderDate) {
		super();
		this.orderid = orderid > 0 ? orderid : null;
		this.orderDate = orderDate;
	}

	public void addItem(LineItem lineItem) {
		items.add(lineItem);
	}

	public int getOrderid() {
		return orderid;
	}

	public Date getOrderDate() {
		return orderDate;
	}

	public List<LineItem> getItems() {
		return items;
	}

	public double calcCheckOutCost() {
		double checkoutcost = 0;
		for (LineItem item : items) {
			checkoutcost += item.costofLineItem();
		}
		return checkoutcost;
	}
}

File: DotNET/C#/ShoppingCartApp/ShoppingCartApp/Order.cs
using System;
using System.Collections.Generic;

namespace ShoppingCartApp
{
    class Order
    {
        private int _orderid;
        private DateTime _orderDate;
        private List<LineItem> _items;

        public Order(int orderid, DateTime orderDate)
        {
            _orderid = orderid;
            _orderDate = orderDate;
            _items = new List<LineItem>();
        }

        public void AddItem(LineItem lineitem)
        {
            _items.Add(lineitem);
        }

        public int OrderId
        {
            get
            {
                return _orderid;
            }
        }

        public DateTime OrderDate
        {
            get
            {
                return _orderDate;
            }
        }

        public List<LineItem> Items
        {
            get
            {
                return _items;
            }
        }

        public double CalculateCheckOutCost()
        {
            double checkoutcost = 0;
            foreach (LineItem item in _items)
            {
                checkoutcost += item.costofLineItem();
            }

            return checkoutcost;
        }

    }
}
